Gems/ScriptAutomation/Code/Source/Utils/Utils.cpp

```diff
@@ -0,0 +1,108 @@
+#include <AzCore/IO/Path/Path.h>
+
+
+namespace ScriptAutomation
+{
+    namespace Utils
+    {
+        void ResizeClientArea(uint32_t width, uint32_t height)
+        {
+            AzFramework::NativeWindowHandle windowHandle = nullptr;
+            AzFramework::WindowSystemRequestBus::BroadcastResult(
+                windowHandle,
+                &AzFramework::WindowSystemRequestBus::Events::GetDefaultWindowHandle);
+
+            AzFramework::WindowSize clientAreaSize = {width, height};
+            AzFramework::WindowRequestBus::Event(windowHandle, &AzFramework::WindowRequestBus::Events::ResizeClientArea, clientAreaSize);
+            AzFramework::WindowSize newWindowSize;
+            AzFramework::WindowRequestBus::EventResult(newWindowSize, windowHandle, &AzFramework::WindowRequests::GetClientAreaSize);
+            AZ_Error("ResizeClientArea", newWindowSize.m_width == width && newWindowSize.m_height == height,
+                "Requested window resize to %ux%u but got %ux%u. This display resolution is too low or desktop scaling is too high.",
+                width, height, newWindowSize.m_width, newWindowSize.m_height);
+        }
+
+        bool SupportsToggleFullScreenOfDefaultWindow()
+        {
+            return AzFramework::NativeWindow::CanToggleFullScreenStateOfDefaultWindow();
+        }
+
+        void ToggleFullScreenOfDefaultWindow()
+        {
+            AzFramework::NativeWindow::ToggleFullScreenStateOfDefaultWindow();
+        }
+
+        AZStd::string GetScreenshotsPath(bool resolvePath)
+        {
+            AZStd::string path = "@user@/scriptautomation/screenshots/";
+
+            if (resolvePath)
+            {
+                path = Utils::ResolvePath(path);
+            }
+
+            return path;
+        }
+
+        AZStd::string GetProfilingPath(bool resolvePath)
+        {
+            AZStd::string path = "@user@/scriptautomation/profiling/";
+
+            if (resolvePath)
+            {
+                path = Utils::ResolvePath(path);
+            }
+
+            return path;
+        }
+
+
+        AZStd::string ResolvePath(const AZStd::string& path)
+        {
+            char resolvedPath[AZ_MAX_PATH_LEN] = {0};
+            AZ::IO::FileIOBase::GetInstance()->ResolvePath(path.c_str(), resolvedPath, AZ_MAX_PATH_LEN);
+            return resolvedPath;
+        }
+
+        bool IsFileUnderFolder(AZStd::string filePath, AZStd::string folder)
+        {
+            AzFramework::StringFunc::Path::Normalize(filePath);
+            AzFramework::StringFunc::Path::Normalize(folder);
+
+            AZStd::to_lower(filePath.begin(), filePath.end());
+            AZStd::to_lower(folder.begin(), folder.end());
+            auto relativePath = AZ::IO::FixedMaxPath(filePath.c_str()).LexicallyRelative(folder.c_str());
+            if (!relativePath.empty() && !relativePath.Native().starts_with(".."))
+            {
+                return true;
+            }
+            return false;
+        }
```

Comment:
The Path API has a function for determining if one path is relative to another.
Furthermore it doesn't require any memory allocations to perform that operation.
It is [PathView::IsRelativeTo](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h#L255-L256) function.

Furthermore the check can be made case-insensitive on all platforms by expliciting constructing the PathView with the AZ::IO::WindowsPathSeparator per the [documentation](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h#L565-L576)

```suggestion
        bool IsFileUnderFolder(AZ::IO::PathView filePath, AZ::IO::PathView folder)
        {
            // To make sure the check is case-insensitive on non-Windows platforms create a copy of the PathView using AZ::IO::WindowsPathSeparator
            return AZ::IO::PathView(filePath.Native(), AZ::IO::WindowsPathSeparator).IsRelativeTo(AZ::IO::PathView(folder.Native(), AZ::IO::WindowsPathSeparator));
        }
```

That being said this utility function isn't very useful if it is only a single line.

Reply:
Why cast to a pathview if the filepath is already a pathview?

Reply:
So bit of a background story.
The default path separator value for all the `AZ::IO::`Path classes defaults to the OS path separator.
So on Windows it defaults to `\` and everywhere else it defaults to the POSIX separator of `/`.
Well that path separator determines if the the path operations are case-sensitive if it uses the POSIX path separator.
When the Windows path separator is used the path operations are case-insensitive.

So we are explicitly constructing a PathView with the WindowsPathSeparator, so that when this code runs on Android, iOS, Linux and MacOS, it still performs a case-insensitive check for a relative path.

But now that I think about it.
The `@user@` folder isn't a product folder in the Asset Cache , so the checks should be case-sensitive on non-Windows platforms.

In which case, we should use the simpler construct of 
```
return filePath.IsRelativeTo(folder);
```

Reply:
Or alternatively we can skip the need for the function altogether and just use a lambda to reflect the method directly to the BehaviorContext
i.e
```
behaviorContext->Method("IsFileUnderFolder, [](AZStd:;string_view filePath, AZStd::string_view folder)
{
    return AZ::IO::PathView(filePath).IsRelativeTo(folder);
});

Reply:
It's also used in the PrepareForScreenCapture function.

Reply:
There we could just call IsRelativeTo directly then as well
```
if (!AZ::IO::PathView(Utils::ResolvePath(path)).IsRelativeTo(Utils::GetScreenshotsPath(true)))
```